src/rules/maxClassesPerFileRule.ts

```diff
@@ -56,19 +69,23 @@ export class Rule extends Lint.Rules.AbstractRule {
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
         const argument = this.ruleArguments[0] as number;
         const maxClasses = isNaN(argument) || argument > 0 ? argument : 1;
-        return this.applyWithFunction(sourceFile, walk, { maxClasses });
+        return this.applyWithFunction(sourceFile, walk, {
+            excludeClassExpressions: this.ruleArguments.indexOf(OPTION_INCLUDE_CLASS_EXPRESSIONS) !== -1,
+            maxClasses,
+        });
     }
 }
 
-interface Options {
-    maxClasses: number;
-}
-
 function walk(ctx: Lint.WalkContext<Options>): void {
-    const { sourceFile, options: { maxClasses } } = ctx;
+    const { sourceFile, options: { maxClasses, excludeClassExpressions } } = ctx;
     let classes = 0;
     return ts.forEachChild(sourceFile, function cb(node: ts.Node): void {
-        if (isClassLikeDeclaration(node)) {
+        if (
+            isClassLikeDeclaration(node) &&
+            (excludeClassExpressions
+                ? !isClassExpression(node)
+                : true)
```

Comment:
could also be `isClassDeclaration(node) || (!excludeClassExpression && isClassExpression(node))`
It's up to you which one you prefer.